### PR TITLE
fix: copy basic plugin properties onto the wrapper

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description
Copy basic plugin properties to prevent breaking older basic plugins

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Unit Tests updated or fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/js/plugin.js [hunk 1]
```diff
@@ -312,6 +312,9 @@ class Plugin {
     if (name !== BASE_PLUGIN_NAME) {
       if (Plugin.isBasic(plugin)) {
         Player.prototype[name] = createBasicPlugin(name, plugin);
+        Object.keys(plugin).forEach(function(prop) {
+          Player.prototype[name][prop] = plugin[prop];
+        });
```
Comment (misteroneill) [inline src/js/plugin.js]: I feel like this belongs in the `createBasicPlugin` function.

Comment (BrandonOCasey) [inline src/js/plugin.js]: I thoughts that too at first, but `Player.prototype[name]` does not exist until we get the return value from `createBasicPlugin`

Comment (BrandonOCasey) [inline src/js/plugin.js]: If we really want it in createBasicPlugin, then we have to do the `Player.prototype[name]` assign in there as well. I don't see any problems with that. Thoughts?

Comment (gkatsev) [inline src/js/plugin.js]: couldn't it just copy the props from `plugin` onto the `instance` that `createBasicPlugin` returns`?

Comment (gkatsev) [inline src/js/plugin.js]: `createBasicPlugin` is a function that [returns a function](https://github.com/videojs/video.js/blob/afa747c2fdeca36aa971f4705dc121f94549daf4/src/js/plugin.js#L91). It's a bit confusing.
But `instance` is inside the returned function, you'd want to do the property copying in the outer arrow function instead.

Comment (gkatsev) [inline src/js/plugin.js]: I almost thought just now that `createBasicPlugin` initialized basic plugins when called, but it doesn't do that.

Comment (BrandonOCasey) [inline src/js/plugin.js]: yeah calling the function that `createBasicPlugin` returns on player is actually what initializes. The question is do we want to copy the methods that are on the plugin right away (this is how it used to work in 5.x)? or only after the plugin has been initialized. 

Examples:

After initialization copy properties
```js
var basic = () => {};
basic.VERSION = '9.9.9';
videojs.registerPlugin('basic', basic);
...
console.log(player.basic.VERSION);
// undefined
player.basic();

console.log(player.basic.VERSION);
// 9.9.9
```

Before initialization copy properties
```js
var basic = () => {};
basic.VERSION = '9.9.9';
videojs.registerPlugin('basic', basic);
...
console.log(player.basic.VERSION);
// '9.9.9'
```

Comment (gkatsev) [inline src/js/plugin.js]: I think we want to mimic current behavior, which means that we want to copy it before init and the plugin would need to handle things after init, like they do now.

**Comment by gkatsev** [comment]
Tests are fine except that IE8 is being weird on browserstack.

**Pull request merged**